commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Activity.kt

```diff
@@ -713,7 +713,8 @@ fun BaseSimpleActivity.deleteFilesBg(files: List<FileDirItem>, allowDeleteFolder
             }
 
             val recycleBinPath = firstFile.isRecycleBinPath(this)
-            if (canManageMedia() && !recycleBinPath) {
+            val containsNoMedia = firstFile.path.doesThisOrParentHaveNoMedia(HashMap(), null)
+            if (canManageMedia() && !recycleBinPath && !containsNoMedia) {
```

Comment:
Lets avoid calling containsNoMedia when it isnt necessary, it can often be slow. I guess the best way (not the way we do it mostly) is just removing the local variable and changing it to
if (canManageMedia() && !recycleBinPath && !firstFile.path.doesThisOrParentHaveNoMedia(HashMap(), null)) {

It could be solved other ways too, but there is an else statement too so it could be messy.

Reply:
done :+1: